newIDE/app/src/ExpressionAutocompletion/index.js

```diff
@@ -207,17 +220,77 @@ const getAutocompletionsForObject = function(
     ...filteredObjectsList.map(({ object }) => ({
       kind: 'Object',
       completion: object.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       object,
       addDot,
     })),
     ...filteredGroupsList.map(({ group }) => ({
       kind: 'Object',
       completion: group.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       addDot,
     })),
   ];
 };
 
+const getAutocompletionsForText = function(
+  expressionAutocompletionContext: ExpressionAutocompletionContext,
+  completionDescription: gdExpressionCompletionDescription
+): Array<ExpressionAutocompletion> {
+  const prefix: string = completionDescription.getPrefix();
+  const type: string = completionDescription.getType();
+  //const objectName: string = completionDescription.getObjectName();
+  const {
+    globalObjectsContainer,
+    objectsContainer,
+  } = expressionAutocompletionContext;
+
+  let autocompletionTexts: string[] = [];
+  if (type === 'layer') {
+    const layout = (objectsContainer: gdLayout);
```

Comment:
Same as the project: we should be able to pass the `layout` when calling `getAutocompletionsFromDescriptions ` and have it passed down thanks to the context. You can find it into the `scope` (of type `EventsScope`) in `GenericExpressionField` :) 

But we have to be even more cautious! It's possible that we're in a context where we don't have a layout (so the type will be a `?gdLayout`), for example when editing a function.